teamcity: failed tests on master: testrace/TestRaftRemoveRace

The following tests appear to have failed:

[#36879](https://teamcity.cockroachdb.com/viewLog.html?buildId=36879):

```
--- FAIL: testrace/TestRaftRemoveRace (0.005s)
	client_test.go:927: [n1,s1,r1/1:/M{in-ax}]: change replicas aborted due to failed preemptive snapshot: range=1: remote declined snapshot: 

```
Please assign, take a look and update the issue accordingly.

```
~ make stress PKG=./storage/ TESTS=TestRaftRemoveRace
...
--- FAIL: TestRaftRemoveRace (0.40s)
       	client_test.go:930: [n1,s1,r1/1:/M{in-ax}]: change replicas aborted due to failed preemptive snapshot: range=1: remote declined snapshot:
FAIL


ERROR: exit status 1

10 runs completed, 1 failures, over 0s
FAIL
```

In other words, this test fails almost immediately under stress. I'm taking a look.

--- a/pkg/storage/store.go
+++ b/pkg/storage/store.go
@@ -2087,7 +2087,7 @@ func (s *Store) Descriptor() (*roachpb.StoreDescriptor, error) {
 	}, nil
 }
 
-// deadReplicas returns a list of all the dead replicas on the store.
+// deadReplicas returns a list of all the corrupt replicas on the store.
 func (s *Store) deadReplicas() roachpb.StoreDeadReplicas {
 	// We can't use a storeReplicaVisitor here as it skips destroyed replicas.
 	// Similar to in the storeReplicaVisitor, make a copy of the current
@@ -2105,11 +2105,11 @@ func (s *Store) deadReplicas() roachpb.StoreDeadReplicas {
 	var deadReplicas []roachpb.ReplicaIdent
 	for _, r := range replicas {
 		r.mu.Lock()
-		destroyed := r.mu.destroyed
+		corrupted := r.mu.corrupted
 		desc := r.mu.state.Desc
 		r.mu.Unlock()
 		replicaDesc, ok := desc.GetReplicaDescriptor(s.Ident.StoreID)
-		if ok && destroyed != nil {
+		if ok && corrupted {
 			deadReplicas = append(deadReplicas, roachpb.ReplicaIdent{
 				RangeID: desc.RangeID,
 				Replica: replicaDesc,

